internal/core/permission_test.go

```diff
@@ -18,7 +18,7 @@ var Queries = map[string]allowed.IsAllowed{
 func TestDispatchNotFound(t *testing.T) {
 	core.Queries = Queries
 	p := core.NewPermissionService(nil)
-	result, addition, err := p.IsAllowed("", 0, nil)
+	result, addition, err := p.IsAllowed(nil, "", 0, nil)
```

Comment:
You should never use a nil-context. In this place, `context.Background()` is probably the right choice.

In the doc is says:

https://golang.org/pkg/context/
> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use.